lib/fastlane/plugin/json/actions/download_json_action.rb

```diff
@@ -9,13 +9,25 @@ class DownloadJsonAction < Action
       def self.run(params)
         json_url = params[:json_url]
         @is_verbose = params[:verbose]
+        username = params[:username]
+        password = params[:password]
+
         uri = URI(json_url)
         print_params(params) if @is_verbose
         puts("Downloading json from #{uri}") if @is_verbose
 
         begin
-          response = Net::HTTP.get_response(uri)
-          JSON.parse(response.body, symbolize_names: true)
+          Net::HTTP.start(uri.host, uri.port,
+                          use_ssl: uri.scheme == 'https',
+                          verify_mode: OpenSSL::SSL::VERIFY_NONE) do |http|
+
+            request = Net::HTTP::Get.new(uri.request_uri)
+            if !username.to_s.empty? && !password.to_s.empty?
```

Comment:
Is it possible you forgot to add the code that adds the auth to the request?

Reply:
> Is it possible you forgot to add the code that adds the auth to the request?

Yeah, my mistake when copying code from the old PR to the new one. I updated it. PTAL